src/main/scala/tech/cryptonomic/conseil/util/DatabaseUtil.scala

```diff
@@ -102,11 +102,12 @@ object DatabaseUtil {
         * @param columns     parameter containing columns which chich are being used in query
         * @return new SQLActionBuilder containing limit statement
         */
-      def addGroupBy(aggregation: Option[Aggregation], columns: List[String]): SQLActionBuilder = {
-        aggregation.fold(action) {
-          aggregates =>
-            val cols = columns.filterNot(_ == aggregates.field)
-          concatenateSqlActions(action, makeGroupBy(cols))
+      def addGroupBy(aggregation: List[Aggregation], columns: List[String]): SQLActionBuilder = {
+        val aggregationFields = aggregation.map(_.field).toSet
```

Comment:
You can move that `val` to be inside `else` statement.

Reply:
ok, done